src/hooks/useUpdateSetting.jsx

```diff
@@ -7,19 +7,24 @@ import { useUserStore } from "@/store/useUserStore";
 const useUpdateSetting = (type) => {
   const { settings, setUserSettings } = useUserStore();
 
-  const updateSetting = async () => {
+  const updateSetting = async (value) => {
     const userId = localStorage.getItem("userId");
     if (!userId) {
       return;
     }
 
-    const updatedSettings = {
-      ...settings,
-      [type]: !settings[type],
-    };
-
-    if (type === SETTING_TYPES.AD_DETECT && !updatedSettings.isAdDetect) {
-      updatedSettings.isReturnChannel = false;
+    const updatedSettings = { ...settings };
```

Comment:
```suggestion
    const updatedSettings = { };
```
앞서 제안드린 zustand setUserSettings를 spread (...state.settings) 방식으로 변경하면
이 부분에서도 전체 settings를 spread 할 필요 없이 변경된 값만 담아도 되기 때문에
`const updatedSettings = {};`와 같이 간단히 초기화해도 될 것 같아요!